### PR TITLE
Update python to 3.12 in env

### PR DESCRIPTION
**Pull request by aknierim**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: toolbox
 channels:
   - conda-forge
 dependencies:
-  - python=3.11
+  - python=3.12
   - pip
   - matplotlib
   - numpy
```
Comment (SepplL) [inline environment.yml:8]: Do we also want to move to `numpy 2` in the same step?

Comment (aknierim) [reply environment.yml:8]: NumPy is already automatically installed as the latest version (2.1.1)

Comment (SepplL) [reply environment.yml:8]: I'm aware of that. That's why I'm asking - do we want to leave it open?

Comment (aknierim) [reply environment.yml:8]: I mean, we can set it to `>=2.0` or `~=2.0`... I wouldn't mind :+1:

Comment (SepplL) [reply environment.yml:8]: If everything works, I'm fine with it. Just recently had the experience with `numpy 2` to be lacking some important "legacy" functions depending on the project.
But if everything we need is also supported, then I'm fine with bumping the version.

Comment (aknierim) [reply environment.yml:8]: Notebook seems to run fine with NumPy 2

**Review by SepplL — APPROVED** [review]

**Pull request merged**